modules/createHistory.js

```diff
@@ -49,7 +54,9 @@ function createHistory(options={}) {
     if (location) {
       listener(location);
     } else {
-      updateLocation(getCurrentLocation());
+      var locationReplacement = getCurrentLocation();
+      replaceState(locationReplacement.state, locationReplacement.pathname + locationReplacement.search);
```

Comment:
I think I'd like to avoid using `replaceState` here if we can. As it is currently implemented, `replaceState` will run through `confirmTransition`, which I think we want to avoid on the initial page load. Also, BrowserHistory does a full page refresh in browsers that don't support HTML5 history when you call `replaceState`, so it will leave those users in an infinite refresh cycle. Instead, I think we can simply update the key of a history inside each implementation's `getCurrentLocation` function.